### PR TITLE
try migrating pull-kubernetes-verify to prow using docker-in-docker

### PR DESCRIPTION
**Pull request by BenTheElder**
This will allow us to start testing migrating `pull-kubernetes-verify` to Prow using our own image with docker-in-docker support derived from the existing image used on Jenkins.

/area images
/area jobs

### CODE REVIEW

**Comment by BenTheElder** [comment]
/hold
pushing then adding a test job
/cc @krzyzacy

File: images/prow-test-image/Dockerfile-1.9 [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Copyright 2017 The Kubernetes Authors.
```
Comment (BenTheElder) [inline images/prow-test-image/Dockerfile-1.9]: This is jenkins/test-images/Dockerfile-1.9 but with docker-in-docker support and a bootstrap runner that runs `service docker start` before bootstrap.

**Comment by krzyzacy** [comment]
`I-Dont-Like-Docker-But-`
I think you can simplify the Makefile to do `docker build` and `docker push`, drop all those version related stuff?

File: images/prow-test-image/Makefile [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Copyright 2016 The Kubernetes Authors.
```
Comment (krzyzacy) [inline images/prow-test-image/Makefile]: bikeshed: 2017

File: images/prow-test-image/runner [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+set -o nounset
+set -o pipefail
+
+service docker start
+git clone https://github.com/kubernetes/test-infra
+./test-infra/jenkins/bootstrap.py \
+    --job=${JOB_NAME} \
+    --service-account=${GOOGLE_APPLICATION_CREDENTIALS} \
+    --upload='gs://kubernetes-jenkins/logs' \
+    "$@"
```
Comment (krzyzacy) [inline images/prow-test-image/runner]: bikeshed: new line

**Comment by BenTheElder** [comment]
> I think you can simplify the Makefile to do docker build and docker push, drop all those version related stuff?

If this works out well we will still need to support branched versions right?

File: prow/config.yaml [hunk 4]
```diff
@@ -1556,6 +1556,52 @@ presubmits:
...
+        secret:
+          secretName: service-account
+      - name: cache-ssd
+        hostPath:
+          path: /mnt/disks/ssd0
+      - name: docker-graph
+        hostPath:
+          path: /mt/disks/ssd0/docker-graph
+      - name: var-lib-docker
+        emptyDir: {}
```
Comment (krzyzacy) [inline prow/config.yaml]: what's this `emptyDir`? Probably you need to add it to https://github.com/kubernetes/test-infra/blob/master/prow/kube/types.go#L77-L83

File: prow/config.yaml [hunk 5]
```diff
@@ -1556,6 +1556,52 @@ presubmits:
     context: pull-kubernetes-verify
     rerun_command: "/test pull-kubernetes-verify"
     trigger: "(?m)^/test( all| pull-kubernetes-verify),?(\\s+|$)"
+  - name: pull-kubernetes-verify-prow
```
Comment (krzyzacy) [inline prow/config.yaml]: security version :-)

File: prow/config.yaml (hunk 4 above)
Comment (BenTheElder) [reply prow/config.yaml]: https://kubernetes.io/docs/concepts/storage/volumes/#emptydir

File: scenarios/kubernetes_verify.py [hunk 6]
```diff
@@ -108,5 +117,8 @@ def main(branch, script, force):
         '--script',
         default='./hack/jenkins/test-dockerized.sh',
         help='Script in kubernetes/kubernetes that runs checks')
+    PARSER.add_argument(
+        '--jenkins', action='sture_true', help='Force Jenkins mode'
```
Comment (krzyzacy) [inline scenarios/kubernetes_verify.py]: store_true :-)

File: prow/config.yaml (hunk 4 above)
Comment (BenTheElder) [reply prow/config.yaml]: we might not need this eventually, but I don't know if our hosts use for the docker images and the path this is mounted to needs to me something else, so this volume is used to map in a tempdir to /var/lib/docker

**Comment by krzyzacy** [comment]
> If this works out well we will still need to support branched versions right?

We probably want to do the kubekins-e2e format rather than exploding 5x Dockerfiles?

**Comment by BenTheElder** [comment]
> We probably want to do the kubekins-e2e format rather than exploding 5x Dockerfiles?

Delete the old ones? There are some differences in version though, I think we can bikeshed this more after we get a job to work, we may also be able to avoid some of the volume stuff, but for now I want to minimize the differences so that it's easier to debug getting the job to run.

File: images/prow-test-image/Makefile (hunk 2 above)
Comment (BenTheElder) [reply images/prow-test-image/Makefile]: done

File: images/prow-test-image/runner (hunk 3 above)
Comment (BenTheElder) [reply images/prow-test-image/runner]: done

File: scenarios/kubernetes_verify.py (hunk 6 above)
Comment (BenTheElder) [reply scenarios/kubernetes_verify.py]: done

File: prow/config.yaml (hunk 5 above)
Comment (BenTheElder) [reply prow/config.yaml]: done

File: prow/config.yaml (hunk 4 above)
Comment (krzyzacy) [reply prow/config.yaml]: I mean you need to add it to prow/kube, or prow will not understand it :-)

Comment (BenTheElder) [reply prow/config.yaml]: https://github.com/kubernetes/test-infra/pull/5343

Comment (BenTheElder) [reply prow/config.yaml]: merged, everything redeployed

**Comment by BenTheElder** [comment]
@krzyzacy everything should be good to go, let's see if this works?
I fully intend to come back once this job works and cleanly migrate everything on Jenkins over with more streamlined images once this works.

File: jobs/config.json [hunk 7]
```diff
@@ -9529,13 +9529,24 @@
   "pull-kubernetes-verify": {
     "args": [
       "--branch=${PULL_BASE_REF}",
+      "--jenkins",
```
Comment (krzyzacy) [inline jobs/config.json]: you need to add `--jenkins` to ci-verify jobs as well I think

Comment (krzyzacy) [reply jobs/config.json]: Or maybe change it to `--prow` so you won't mess up with existing jobs

Comment (BenTheElder) [reply jobs/config.json]: hmm yeah, let me flip it to `--prow` for now

Comment (BenTheElder) [reply jobs/config.json]: done

**Comment by BenTheElder** [comment]
/hold
I think jenkins-operator is borked

**Comment by BenTheElder** [comment]
had to rollback some Prow components (unrelated) so we will have to come back to this after we can get them succesfully upgraded in order to use emptyDir.

**Pull request merged**